cupy/_core/_reduction.pyx

```diff
@@ -581,6 +581,10 @@ cdef class _SimpleReductionKernel(_AbstractReductionKernel):
     def __call__(self, object a, axis=None, dtype=None, _ndarray_base out=None,
                  bint keepdims=False):
 
+        if hasattr(a, '__cupy_override_reduction_kernel__'):
```

Comment:
This is currently only working for `SimpleReductionKernel`
There are other more generic Reduction kernels, I wonder if we can easily support them?

Reply:
`__cupy_override_reduction_kernel__` has to be called here, before type checks which rule out `DistributedArray`.
I believe we can support `ReductionKernel` in the same way, by adding this hook in its `__call__` method.